biocrnpyler/dna_construct.py

```diff
@@ -397,12 +403,142 @@ def enumerate_components(self,previously_enumerated=None):
+        rev_con = [a.get_orphan() for a in construct]
+        for rev_part,origpart in zip(rev_con,construct):
+            rev_part.direction = {"forward":"reverse","reverse":"forward"}[origpart.direction]
+        rev_con = rev_con[::-1]
+        rev_con = Construct(rev_con,make_dirless_hash=False,circular = construct.circular)
+        return rev_con
+    @classmethod
+    def rotation_free_hash(cls,construct,return_rotation=False):
+        """calculates a unique circular permutation that is the most alphabetically ordered. Every part is considered as a potential starting
+        point, and the most alphabetically ordered order is then chosen as the best permutation"""
+        def circular_next(part,construct):
+            if(isinstance(part,int)):
+                part_pos = part
+            else:
+                part_pos = part.position+1
+            part_pos = part_pos%(len(construct))
+            return construct[part_pos]
+        best_partlist = [(None,None)]
+        for part in construct:
+            parthash = Construct.get_partstring(part)
+            if(best_partlist[0][0] is None):
+                best_partlist = [(parthash,part)]
+            elif(parthash > best_partlist[0][0]):
+                best_partlist = [(parthash,part)]
+            elif(parthash == best_partlist[0][0]):
+                test_partlist = [(parthash,part)]
+                testpos = part.position
+
+                while test_partlist[-1][0] == best_partlist[-1][0]:
+                    if(testpos-part.position>= len(construct)):
+                        #this means we tested every part and they would come in at the same alphabetical position
+                        break
+                    testpos += 1
+                    next_testpart = circular_next(testpos,construct)
+                    test_partlist += [(Construct.get_partstring(next_testpart),next_testpart)]
+
+                    if(len(best_partlist)<len(test_partlist)):
+                        #this means we haven't made the hash string for the next part down the line
+                        next_part = circular_next(best_partlist[-1][1],construct)
+                        next_parthash = Construct.get_partstring(next_part)
+                        best_partlist += [(next_parthash,next_part)]
+                if(test_partlist[-1][0] > best_partlist[-1][0]):
+                    best_partlist = test_partlist
+        while(len(best_partlist)<len(construct)):
+            nextpart = circular_next(best_partlist[-1][1],construct)
+            best_partlist += [(Construct.get_partstring(nextpart),nextpart)]
+        if(return_rotation):
+            return Construct.get_partlist_hash(best_partlist),best_partlist[0][1].position
+        else:
+            return Construct.get_partlist_hash(best_partlist)
+    @classmethod
+    def direction_rotation_free_hash(cls,construct,return_transform = False):
```

Comment:
`direction_rotation_free_hash()` is always used with `return_transform=True`. Can you remove the `False` branch?